### PR TITLE
DI-2029: Changes needed for livy DBAPI

### PR DESCRIPTION
**Pull request by gonzalezzfelipe**

### CODE REVIEW

File: hive/Dockerfile [hunk 1]
```diff
@@ -46,6 +46,9 @@ curl -L https://repo1.maven.org/maven2/com/amazonaws/aws-java-sdk-s3/${AWS_VERSI
 curl -L https://repo1.maven.org/maven2/org/apache/hadoop/hadoop-aws/${HADOOP_VERSION}/hadoop-aws-${HADOOP_VERSION}.jar -o ${HADOOP_HOME}/share/hadoop/common/lib/hadoop-aws.jar && \
 curl -L https://repo1.maven.org/maven2/com/amazonaws/aws-java-sdk-dynamodb/${AWS_VERSION}/aws-java-sdk-dynamodb-${AWS_VERSION}.jar -o ${HADOOP_HOME}/share/hadoop/common/lib/aws-java-sdk-dynamodb-.jar
 
+# Install Iceberg jar
+RUN curl -L https://repo1.maven.org/maven2/org/apache/iceberg/iceberg-hive-runtime/0.12.0/iceberg-hive-runtime-0.12.0.jar -o ${HADOOP_HOME}/share/hadoop/common/lib/iceberg-hive-runtime-0.12.0.jar
```
Comment (viniolivieri) [inline hive/Dockerfile:50]:
```suggestion
RUN curl -L https://repo1.maven.org/maven2/org/apache/iceberg/iceberg-hive-runtime/0.12.1/iceberg-hive-runtime-0.12.1.jar -o ${HADOOP_HOME}/share/hadoop/common/lib/iceberg-hive-runtime-0.12.1.jar
```

File: spark/spark-defaults.conf [hunk 2]
```diff
@@ -39,6 +39,13 @@ spark.hadoop.fs.s3.pathStyleAccess=true
 spark.hadoop.fs.s3.impl=org.apache.hadoop.fs.s3a.S3AFileSystem
 
 spark.sql.orc.impl=hive
+spark.sql.warehouse.dir=hdfs:///user/spark/warehouse
 
 spark.driver.extraClassPath=/opt/hive/lib/*.jar
 spark.executor.extraClassPath=/opt/hive/lib/*.jar
+
+# Iceberg
+spark.jars.packages=org.apache.iceberg:iceberg-spark3-runtime:0.12.0
```
Comment (viniolivieri) [inline spark/spark-defaults.conf:48]:
```suggestion
spark.jars.packages=org.apache.iceberg:iceberg-spark3-runtime:0.12.1
```

File: hive/conf/hive-site.xml [hunk 3]
```diff
@@ -1,18 +1,36 @@
...
+
+  <property>
+    <name>hive.server2.thrift.url</name>
+    <value>thrift://hive-server:10000</value>
+  </property>
+
+  <property>
+    <name>hive.metastore.connect.retries</name>
+    <value>15</value>
+  </property>
```
Comment (viniolivieri) [inline hive/conf/hive-site.xml:34]: These configuration aren't created in this file `hadoop-hive.env`?

At least the URI for the metastore is defined there.

**Review by viniolivieri — APPROVED** [review]
One question and I would apply already the minor patch of Iceberg that fixes a few bugs.

File: hive/conf/hive-site.xml (hunk 3 above)
Comment (gonzalezzfelipe) [reply hive/conf/hive-site.xml:34]: I guess yes, but both spark and livy use the `hive-site.xml` to connect to hive, that's why I added it here

File: livy/Dockerfile [hunk 4]
```diff
@@ -23,16 +23,17 @@ WORKDIR /src/livy
...
-    wget https://downloads.apache.org/incubator/livy/${LIVY_VERSION}/apache-livy-${LIVY_VERSION}-bin.zip  && \
+    wget --no-check-certificate --no-proxy https://artifacts.jampp.com.s3.amazonaws.com/livy/apache-livy-${LIVY_VERSION}-bin.zip  && \
     unzip apache-livy-${LIVY_VERSION}-bin.zip && \
     rm apache-livy-${LIVY_VERSION}-bin.zip && \
     apt-get --purge remove -y wget unzip && \
     apt-get autoremove -y && \
-	apt-get clean && \
-	rm -rf /var/lib/apt/lists/*
+  apt-get clean && \
+  rm -rf /var/lib/apt/lists/*
```
Comment (Rcharriol) [inline livy/Dockerfile:32]: Weird identation is weird

File: spark/Dockerfile [hunk 5]
```diff
@@ -27,14 +27,17 @@ ENV INIT_DAEMON_BASE_URI=http://identifier/init-daemon
...
-    tar -xvzf spark-${SPARK_VERSION}-bin-without-hadoop.tgz && \
-    mv spark-${SPARK_VERSION}-bin-without-hadoop ${SPARK_HOME} && \
-    rm spark-${SPARK_VERSION}-bin-without-hadoop.tgz && \
-    apt-get --purge remove -y wget && \
+    wget https://archive.apache.org/dist/spark/spark-${SPARK_VERSION}/spark-${SPARK_VERSION}-bin-hadoop3.2.tgz && \
+    tar -xvzf spark-${SPARK_VERSION}-bin-hadoop3.2.tgz && \
+    mv spark-${SPARK_VERSION}-bin-hadoop3.2 ${SPARK_HOME} && \
+    rm spark-${SPARK_VERSION}-bin-hadoop3.2.tgz && \
 	apt-get clean && \
 	rm -rf /var/lib/apt/lists/*
```
Comment (Rcharriol) [inline spark/Dockerfile:35]: From weird under identated, to over compensate in the next opportunity, weird is weird

**Review by Rcharriol — APPROVED** [review]

**Pull request merged**